### ISSUE TITLE
Gradle support for secondary versions.props file

### ISSUE THREAD

**Issue by tpittmanVS** [renovatebot/renovate#8017]
In addition to the typical build.gradle file, we store version info in versions.props as described here https://github.com/palantir/gradle-consistent-versions. We need versions.props tracked and incremented in the same way as renovatebot does for build.gradle

Is this feature currently supported or in the product roadmap?

**Comment by rarkins** [comment]
I know we certainly don't support the versions.lock part, so the answer is probably No. Can you provide an example repository and elaborate on the requirements further? e.g. is this completely defined by the palantir plugin only or is that also leveraging some other plugins or Gradle concepts Eg for the lock file?

**Comment by tpittmanVS** [comment]
Hey Rhys, thanks for your quick response. I think this simple example captures our use case: https://github.com/palantir/conjure-java-example. I believe gradle substitutes values from top-level `versions.props` when evaluating `build.gradle`.  

As far as implementation details, I'm not too sure as I'm new to Gradle. I know the whole consistent versions/versions.prop framework does use Gradle Constraints. So to my eye, Gradle and Gradle Wrapper (gradlew) seem to be the only dependencies, but I have never set it up personally  

For your own information, we use this framework across all our repos, and no one in the market seems to support it (dependabot, renovatebot, snyk, etc). We are considering whether it might be worth it to submit a PR to implement this functionality ourselves

**Comment by viceice** [comment]
I think we should add a separate manager for handling that file, as it's easy to parse and update by renovate.
The only part we need todo is running gradle to update the lockfile. This manager can support lockfile only. 🙃 

Maybe we can name it  `gradle-consistent-versions`. I'm very interested in this, as it would make dependency updating for renovate  and gradle very easy. 🎉

**Comment by HonkingGoose** [comment]
@rarkins I'm not sure if you're still waiting on a reproduction here? I was going to toggle the stale-bot on this, but I'll let you do that if needed.

**Comment by janhoy** [comment]
Any status on this? I plan to attempt making a new manager. It would need to
- Parse versions.props and versions.lock
- Filter versions.lock against versions.props, leaving only the entries that match at least one props line (may be wildcard expr), leaving our direct and not transitive dependencies.
- Parse versions and lookup upgrades
- The PR needs to be made against versions.props, and run `gradlew —write-locks` before committing

**Comment by viceice** [comment]
> Any status on this? I plan to attempt making a new manager. It would need to
> - Parse versions.props and versions.lock
> - Filter versions.lock against versions.props, leaving only the entries that match at least one props line (may be wildcard expr), leaving our direct and not transitive dependencies.
> - Parse versions and lookup upgrades
> - The PR needs to be made against versions.props, and run `gradlew —write-locks` before committing

no new manager. should be implemented inside Gradle manager.

**Comment by viceice** [comment]
Blocked by
- https://github.com/renovatebot/renovate/pull/18484

**Comment by janhoy** [comment]
Is this unblocked due to https://github.com/renovatebot/renovate/pull/18663 ?

**Comment by rarkins** [comment]
Yes no longer blocked

**Comment by janhoy** [comment]
Ok, I'm attempting a PR to add support for this plugin.

**Comment by janhoy** [comment]
I now have a working branchthat produces correct PRs for this plugin, see https://github.com/janhoy/gradle-consistent/pulls

I'll do some cleanups and publish a PR...

**Comment by janhoy** [comment]
Another semi-related question:

In one project that I plan to start using renovate with this feature, the automated PR validation process also includes validating SHA-sum and licenses for all dependencies, i.e. we "lock" not only the dependency version, but also the dependency JAR SHA-sum in the repository.

So when renovate opens a PR, it will patch `versions.props`, update `versions.lock` (through `--write-locks`), but we also need it to run an arbitrary gradle task to compute new checksums. I see from the docs that there is `postUpgradeTasks` for self-hosted instances of renovate, due to security reasons of course. We'd like to use the official version, but I cannot find a way to specify an additional gradle task to run as part of the pre-commit update.

Before I open a new feature request issue, I thought I'd discuss it in context here. I'm thinking there could be a way for a repo to specify in `renovate.json` an additional gradle task, e.g. 

```json
{
  "gradle": {
    "postUpgradeGradleTasks": ["task1", "task2"], 
    "postUpgradeFileFilters": ["shasums.txt"],
  }
}
```

Techinically we already run `./gradlew --write-locks` or update-locks, so it could be appended to the same command, but would there be security concerns or other concerns? If you will consider it, I'll open a new issue for it.

**Comment by viceice** [comment]
yes, there is a concern, as the task can manipulate the git directory/ hooks or the shared cache folder. so we don't like to allow arbitrary commands.
